pkg/build/admission/defaults/admission.go

```diff
@@ -71,6 +71,28 @@ func (b BuildDefaults) applyPodDefaults(pod *kapi.Pod) {
 	for k, v := range b.config.Annotations {
 		addDefaultAnnotations(k, v, pod.Annotations)
 	}
+
+	// Apply default resources
+	defaultResources := b.config.Resources
+	if len(pod.Spec.Containers[0].Resources.Limits) == 0 {
+		pod.Spec.Containers[0].Resources.Limits = kapi.ResourceList{}
+	}
+	for name, value := range defaultResources.Limits {
+		if _, ok := pod.Spec.Containers[0].Resources.Limits[name]; !ok {
+			glog.V(5).Infof("Setting default resource limit %s for pod %s/%s to %s", name, pod.Namespace, pod.Name, value)
+			pod.Spec.Containers[0].Resources.Limits[name] = value
+		}
+	}
+	if len(pod.Spec.Containers[0].Resources.Requests) == 0 {
```

Comment:
same as above (and also this code was copied from what applyBuildDefaults is doing).